code/modules/mechs/components/_components.dm

```diff
@@ -101,27 +109,91 @@
 	to_chat(user, SPAN_NOTICE(" - Integrity: <b>[round((((max_damage - total_damage) / max_damage)) * 100)]%</b>" ))
 
 /obj/item/mech_component/attackby(obj/item/I, mob/living/user)
-	if(I.use_tool(user, src, WORKTIME_INSTANT, QUALITY_SCREW_DRIVING, FAILCHANCE_ZERO))
-		if(contents.len)
-			//Filter non movables
-			var/list/valid_contents = list()
-			for(var/atom/movable/A in contents)
-				if(!A.anchored)
-					valid_contents += A
-			if(!valid_contents.len)
-				return
-			var/obj/item/removed = pick(valid_contents)
-			if(!(removed in contents))
-				return
-			if(eject_item(removed, user))
-				update_components()
-		else
-			to_chat(user, SPAN_WARNING("There is nothing to remove."))
+	var/list/usable_qualities = list(QUALITY_PULSING, QUALITY_BOLT_TURNING, QUALITY_WELDING, QUALITY_PRYING, QUALITY_SCREW_DRIVING)
+	var/tool_type = I.get_tool_type(user, usable_qualities, src)
+
+	switch(tool_type)
+		if(QUALITY_SCREW_DRIVING)
+			if(contents.len)
+				//Filter non movables
+				var/list/valid_contents = list()
+				for(var/atom/movable/A in contents)
+					if(!A.anchored)
+						valid_contents += A
+				if(!valid_contents.len)
+					return
+				var/obj/item/removed = pick(valid_contents)
+				if(!(removed in contents))
+					return
+				if(eject_item(removed, user))
+					update_components()
+			else
+				to_chat(user, SPAN_WARNING("There is nothing to remove."))
+
+		if(QUALITY_WELDING)
+			repair_brute_generic(I, user)
+			return
+
+	if(istype(I, /obj/item/stack/cable_coil))
+		repair_burn_generic(I, user)
 		return
+
 	if(istype(I, /obj/item/device/robotanalyzer))
 		to_chat(user, SPAN_NOTICE("Diagnostic Report for \the [src]:"))
 		return_diagnostics(user)
 	return ..()
 
 /obj/item/mech_component/proc/update_components()
 	return
+
+
+/obj/item/mech_component/proc/repair_brute_generic(obj/item/I, mob/user)
+	var/weld_amt = round(max(0, 10, 5 + user.stats.getStat(STAT_MEC) / 3))
+
+	if(brute_damage <= 0)
+		to_chat(user, SPAN_NOTICE("You inspect \the [src] but find nothing to weld."))
+		return
+
+	if(total_damage >= max_damage)
+		to_chat(user, SPAN_WARNING("This part is too damaged to be repaired with a welder!"))
```

Comment:
```suggestion
		to_chat(user, SPAN_WARNING("This part is too damaged to be repaired with \a [I]!"))
```
To be more generic.

Reply:
changed it to "This component is completely destroyed, you can't repair it!" instead of a [thing] otherwise a player might think their [thing] is too weak to repair it when it can't be repaired